documentation/source/triggers.rst

```diff
@@ -2,47 +2,76 @@ Triggers
 
-Simulation Timing
------------------
+    @cocotb.coroutine
+    def coro():
+        print("Some time before the edge")
+        yield RisingEdge(clk)
+        print("Immediately after the edge")
 
-:class:`Timer(time) <.Timer()>`
-    Registers a timed callback with the simulator to continue execution of the coroutine
-    after a specified simulation time period has elapsed.
+Or using the syntax in python 3.5 onwards:
 
+.. code-block:: python3
 
-:class:`.ReadOnly()`
-    Registers a callback which will continue execution of the coroutine when the current simulation timestep moves to the :any:`ReadOnly` phase of the RTL simulator.
-    The :any:`ReadOnly` phase is entered when the current timestep no longer has any further delta steps.
-    This should be a point where all the signal values are stable as there are no more RTL events scheduled for the timestep.
-    The simulator should not allow scheduling of more events in this timestep.
-    Useful for monitors which need to wait for all processes to execute (both RTL and cocotb) to ensure sampled signal values are final.
+    @cocotb.coroutine
+    async def coro():
+        print("Some time before the edge")
+        await RisingEdge(clk)
+        print("Immediately after the edge")
 
+.. _simulator-triggers:
 
-Signal related
---------------
+Simulator Triggers
+------------------
 
-:class:`Edge(signal) <.Edge()>`
-    Registers a callback that will continue execution of the coroutine on any value change of *signal*.
+Signals
+~~~~~~~
 
-:class:`RisingEdge(signal) <.RisingEdge()>`
-    Registers a callback that will continue execution of the coroutine on a transition from ``0`` to ``1`` of *signal*.
+.. autoclass:: cocotb.triggers.Edge
 
-:class:`FallingEdge(signal) <.FallingEdge()>`
-    Registers a callback that will continue execution of the coroutine on a transition from ``1`` to ``0`` of *signal*.
+.. autoclass:: cocotb.triggers.RisingEdge
 
-:class:`ClockCycles(signal, num_cycles) <.ClockCycles>`
-    Registers a callback that will continue execution of the coroutine when *num_cycles* transitions from ``0`` to ``1`` have occured on *signal*.
+.. autoclass:: cocotb.triggers.FallingEdge
+
+.. autoclass:: cocotb.triggers.ClockCycles
+
+
+Timing
+~~~~~~
+
+.. autoclass:: cocotb.triggers.Timer
+
+.. autoclass:: cocotb.triggers.ReadOnly
+
+.. autoclass:: cocotb.triggers.ReadWrite
+
+.. autoclass:: cocotb.triggers.NextTimeStep
 
 
 Python Triggers
 ---------------
 
-:class:`.Event()`
-    Can be used to synchronise between coroutines.
-    Yielding :meth:`.Event.wait()` will block the coroutine until :meth:`.Event.set()` is called somewhere else.
+.. autoclass:: cocotb.triggers.Combine
+
+.. autoclass:: cocotb.triggers.First
+
+.. autoclass:: cocotb.triggers.Join
+    :members: retval
+
+
+Synchronization
+~~~~~~~~~~~~~~~
+
+These are not :class:`Trigger`\ s themselves, but contain methods that can be used as triggers.
```

Comment:
```
You may supply an explicit title and reference target, like in reST direct hyperlinks: :role:`title <target>` will refer to target, but the link text will be title.
```
(https://www.sphinx-doc.org/en/1.5/markup/inline.html#cross-referencing-syntax)

Reply:
This was a deliberate stylistic choice - I'm a fan of `use()`ing variable names as part of longer words in order to make it clear I'm referring to the symbol.

I'm not opposed to changing this if you disagree with that choice.

Reply:
No, no disagreement if this was deliberate.